### ISSUE TITLE
Fabric build fails on iOS with nightly version

### ISSUE THREAD

**Issue by zolbooo** [software-mansion/react-native-screens#1482]
## Description

Latest version of `react-native-screens` from Git (f1a8bd2 at the time of writing this issue) cannot compile because of invalid type in [`RNSScreenStack.mm`](https://github.com/software-mansion/react-native-screens/blob/main/ios/RNSScreenStack.mm#L922). Since React Native 0.69.0-rc.0, `MountingTransactionMetadata` was removed in favor of `MountingTransaction` ([source](https://github.com/facebook/react-native/blob/v0.69.0-rc.0/React/Fabric/Mounting/RCTMountingTransactionObserving.h#L53)) in this [commit](https://github.com/facebook/react-native/commit/3a721f48b138dfa587ffdd63917cc24a95286723).

### Screenshots
--

## Steps To Reproduce

1. Create new app using  RN 0.69
2. Add `react-native-screens` dependency in `package.json`:
```
"react-native-screens": "software-mansion/react-native-screens#f1a8bd2"
```
3. Run `cd ios && USE_FABRIC=1 RCT_NEW_ARCH_ENABLED=1 pod install`
4. Run `npm run ios`

### Expected behavior
Build should not fail

### Actual behavior
Compile error in iOS build

## Reproduction

Build latest version of `react-native-screens` from Git using steps provided above.

## Platform

- [x] iOS
- [ ] Android
- [ ] Web
- [ ] Windows
- [ ] tvOS

## Workflow

- [ ] Managed workflow
- [x] Bare workflow

## Package versions

| package                                | version |
| -------------------------------------- | ------- |
| react-native                           |     0.69.0-rc.6    |
| @react-navigation/native               |    6.0.10     |
| react-native-screens                   |    software-mansion/react-native-screens#f1a8bd2     |
| react-native-gesture-handler           |     4.3.1    |

**Comment by kyle-ssg** [comment]
Is this a duplicate of https://github.com/software-mansion/react-native-screens/issues/1474 perhaps?

**Comment by zolbooo** [comment]
Great, thank you! I will use `patch-package` for now.

**Comment by Mattesackboy** [comment]
Having here the same issue, hoping to see a new version published with this fix soon now that React Native 0.69 is officially released! 🙏🏻
